### PR TITLE
Add ReadRel LocalFiles

### PR DESCRIPTION
**Pull request by baibaichen**
Currently substrait-java doesn't implement `LocalFiles` ReadRel,  this patch adds this missing `Rel`.  

## How to test

Add a new UT `LocalFilesRoundtripTest`

### CODE REVIEW

**Comment by baibaichen** [comment]
cc @jacques-n

**Review by jacques-n — CHANGES_REQUESTED** [review]
This generally looks good. I think it probably makes sense to make each of the formats empty classes instead of enums so that we can introduce options on each of those formats later without having breaking changes. Can you make that change?

**Comment by baibaichen** [comment]
> This generally looks good. I think it probably makes sense to make each of the formats empty classes instead of enums so that we can introduce options on each of those formats later without having breaking changes. Can you make that change?

Sure, let me first verify it in the gluten and see whther it's ok or not, it could take me a while.

File: isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java [hunk 1]
```diff
@@ -112,6 +113,11 @@ public RelNode visit(NamedScan namedScan) throws RuntimeException {
     return relBuilder.scan(namedScan.getNames()).build();
   }
 
+  @Override
+  public RelNode visit(LocalFiles localFiles) throws RuntimeException {
+    throw new UnsupportedOperationException("LocalFiles is not supported");
```
Comment (jacques-n) [inline isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java:118]: Why are you throwing here?

**Review by jacques-n — COMMENTED** [review]
Can you please add roundtrip tests to this?

File: isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java (hunk 1 above)
Comment (baibaichen) [reply isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java:118]: > Why are you throwing here?

Do you  know which Calcite rel to use for `LocalFiles` ?

File: core/src/test/java/io/substrait/type/proto/LocalFilesRoundtripTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import io.substrait.relation.files.ImmutableFileFormat;
+import io.substrait.relation.files.ImmutableFileOrFiles;
+import io.substrait.type.ImmutableNamedStruct;
+import io.substrait.type.Type;
+import io.substrait.type.TypeCreator;
+import java.io.IOException;
+import java.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+public class LocalFilesRoundtripTest {
```
Comment (baibaichen) [inline core/src/test/java/io/substrait/type/proto/LocalFilesRoundtripTest.java:25]: @jacques-n 

Do you mean this?

Comment (jacques-n) [reply core/src/test/java/io/substrait/type/proto/LocalFilesRoundtripTest.java:25]: Yes, apparently missed that on first pass. Thanks

**Review by jacques-n — APPROVED** [review]

**Pull request merged**